### PR TITLE
Add search, image updating, links between pages, and html5 tags

### PR DESCRIPTION
**Pull request by bxw201**
First, for image updating, I ended up going with this schema for the sake of simplicity:
If the user adds no files, the images will stay the same. If they do add files, all the images will become the new files that they uploaded.
I figured that would be the simplest way to go about it without having to do some front end Javascript to hack together some sort of UI. Let me know if you have any other ideas though, I'm 100% down.

Second, we should talk about implementing a `utils.py` soon.

Lastly, right now the UI sends you to the error page when you search something up and no results return. I'm not sure if we want that UX, but I'm not exactly sure what to implement. Let me know what you think.

### CODE REVIEW

**Review by hpdouglass — APPROVED** [review]
Overall looks good! The `{{listing._id}}` is a super nitpick, and I don't think the UX thing is too terrible either, if we wanted to just have it error out in the short term and replace that behavior in another PR. I'll let you decide if you want to merge this PR as-is or make changes first.

**Comment by hpdouglass** [comment]
One more thing, I'm happy to implement a `utils.py` in my next PR. I'm thinking of the following 3 methods that should be implemented:

`get_listing_by_id(id)`: returns listing from database based on given id.
`is_password_valid_for_listing(password, listing)`: checks if password is valid for given listing

File: listing_ask.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 def listing_ask(id):
     if request.method == "GET":
         # Request method is GET, display the form for editing and do not run below code.
-        return render_template('listing_ask.html')
+        return render_template('listing_ask.html', id = id)
```
[the remark this answers is not in the document]
Comment (bxw201) [reply listing_ask.py:10]: Made the change for all files except this one. When the method is GET, we don't yet have access to the listing, so we take the id from the url.

**Comment by bxw201** [comment]
> One more thing, I'm happy to implement a `utils.py` in my next PR. I'm thinking of the following 3 methods that should be implemented:
> 
> `get_listing_by_id(id)`: returns listing from database based on given id. `is_password_valid_for_listing(password, listing)`: checks if password is valid for given listing `is_file_image(filename)`: checks if a file is an allowed image type (`ALLOWED_FILES` or whatever it is called would also be moved)
> 
> Can you think of anything else?

Sounds good to me. I'll add it in if I think of anything.

**Pull request merged**